src/validation.cpp

```diff
@@ -4471,6 +4474,21 @@ void Chainstate::LoadExternalBlockFile(
                     }
                 }
 
+                if (fPruneMode && !fReindex && pblock) {
```

Comment:
Any reason not to just move this up to [after we accept the block](https://github.com/bitcoin/bitcoin/pull/24957/files#diff-97c3a52bc5fad452d82670a7fd291800bae20c7bc35bb82686c2c0a4ea7b5b98L4461)?

Reply:
Yes, I can't move my updates for this PR into that code block because `ActivateBestChain` asserts that `cs_main` is not held.